packages/state/src/lib/react/useValueDebounced.ts

```diff
@@ -0,0 +1,33 @@
+/* eslint-disable prefer-rest-params */
+import { useEffect, useState } from 'react'
+import { useValue } from './useValue'
+
+/**
+ * Extracts the value from a signal and subscribes to it, debouncing the value by the given number of milliseconds.
+ *
+ * @see [[useValue]] for more information.
+ *
+ * @public
+ */
+export function useValueDebounced<Value>(
```

Comment:
it seems like this would be simpler & more versatile if it took any value (ie from `useValue`) instead of calling `useValue` itself, like:
```ts
const rawValue = useValue(...)
const value = useDebounced(rawValue, 100)
```

what do you think?

Reply:
The impetus here stemmed from the comment on the original PR from Steve (albeit in a little bit of a different form): https://github.com/tldraw/tldraw/pull/3764#discussion_r1605922639
It felt like maybe this would be a useful pattern to have for reactors.

Like, I don't disagree, per se, about splitting them up and then composing them. But I think it might be a useful primitive DX-wise to add alongside `useValue`

Reply:
it almost sort of feels like the wrong level for a signia specific thing - like, i could imagine a function that debounces a signal being valuable as a signia primitive? but it feels a little weird to me to have this as a specific signia hook, especially as it forces quite an awkward signature compared to how most hooks-y code looks. happy to leave as is if you're sure tho!

Reply:
forgot to respond to this: can't say i'm _entirely_ sure but i think there's probably room for both ideas! cc @ds300 
i think debouncing a signal as a signia primitive is something we should also add. but also, this composed hook feels like a useful additive thing.

Reply:
happy to merge this as `@internal` for now. Agree that a debounced thing makes sense to offer but i think it would make more sense to be an option for `useValue` and then be implemented by debouncing the effect scheduler, so that only the debounced updates actually trigger rerenders.

Reply:
ah just seeing your comment - i'll mark this as `@internal` in a separate PR 👍